### ISSUE TITLE
Cache miss with AGP 7.4

### ISSUE THREAD

**Issue by GianfrancoMS** [gradle/android-cache-fix-gradle-plugin#416]
At my company, we're trying to update to AGP 7.4 and to the Android Cache Fix Gradle Plugin 2.6.2, but we have encountered a cache miss with the task `compile${variant}JavaWithJavac`. This cache miss did not happen in previous versions.

Versions:
- Gradle 7.5.1
- Kotlin 1.7.21
- AGP 7.4 (previous was 7.3.1)
- Android Cache Fix Gradle Plugin 2.6.2 (previous was 2.6.1)
- JDK 17

<img width="1216" alt="Screenshot 2023-01-14 at 11 55 20" src="https://user-images.githubusercontent.com/22310159/212485107-79a4a7e1-ee4a-42e2-bd7e-86a00fcab503.png">

Let me know if you need a scan dump since my GE instance is behind a VPN.

**Comment by cdsap** [comment]
hi @GianfrancoMS, thanks for reporting this issue. 
I notice the issue is coming from the `jdkImage`. We provided a JdkImageWorkaround for AGP < 7.4 but the workaround is not applied for AGP 7.4 since the fix was provided in AGP 7.4.0-alpha07 (https://issuetracker.google.com/u/1/issues/234820480#comment6)

If you test 2.6.2 with AGP 7.3.1 works right?

We will test again the AGP 7.4 reproducing the JdkImage issue. If is not fixed, we will apply again the workaround in the latest AGP.

**Comment by GianfrancoMS** [comment]
2.6.2 with AGP 7.3.1 works perfectly. It only happens with AGP 7.4.

**Comment by cdsap** [comment]
Hi @GianfrancoMS, I finished the investigation and here is the result:

The existing issue mentioned https://issuetracker.google.com/u/1/issues/234820480 is not reproducible in AGP 7.4, so the workaround is correctly not applied in the AGP 7.4.

Now, why are we seeing this cache miss in the JavaCompile? 
First, I reproduced your issue in a simple project with AGP 7.4 and Android Cache Fix Plugin 2.6.2 executing builds in different Java versions:
![Screen Shot 2023-01-17 at 6 28 12 PM](https://user-images.githubusercontent.com/475733/213067020-b7129124-c998-4294-acd0-84ee24b7fcad.png)
I noticed that the reason is different from one project using 7.3.1:
![Screen Shot 2023-01-17 at 6 29 49 PM](https://user-images.githubusercontent.com/475733/213067297-4074d17c-32c0-4178-8c7a-7286115c6363.png)

We observe that the file properties differences lay on `jrt-fs.jar` for our builds with 7.4 and for 7.3 is the entire folder. 
That means, that Google has fixed correctly the original issue, but still we are seeing differences in the jar.

This is a [known issue](https://github.com/gradle/android-cache-fix-gradle-plugin/issues/341) that [normalization](https://docs.gradle.org/current/userguide/more_about_tasks.html#sec:configure_input_normalization) can fix. To be sure, I compared two builds with different Java versions and collected the jar for each job. Applying a diff on the two folders we have:
![Screen Shot 2023-01-17 at 6 39 55 PM](https://user-images.githubusercontent.com/475733/213068670-9abf5808-38ac-436e-9e31-7ce66c436223.png)
We can notice small differences in the internal folders `jimage` and `decompressor`. 
So applying the normalization:
```
allprojects {
     normalization {
        runtimeClasspath {
            ignore '**/jdk/internal/jimage/decompressor/**'
            ignore '**/jdk/internal/jrtfs/**'
        }
    }
}
```
Fixes the issue:
![Screen Shot 2023-01-17 at 6 43 35 PM](https://user-images.githubusercontent.com/475733/213069238-41a57ef2-25bd-4129-a104-9a225a6696c7.png)


Please let me know if the normalisation configuration works for you, I would recommend verifying the jar differences to better understand what to normalize.

**Comment by GianfrancoMS** [comment]
It's a really weird difference.

![Screenshot 2023-01-18 at 17 53 47](https://user-images.githubusercontent.com/22310159/213312980-f953ae52-b09b-4b0d-bda5-cc929f802e08.png)

CI
- OS: Linux 4.14.77-80.57.amzn2.x86_64 (amd64)
- Java runtime: Private Build OpenJDK Runtime Environment 17.0.5+8-Ubuntu-2ubuntu122.04
- Java VM: Private Build OpenJDK 64-Bit Server VM 17.0.5+8-Ubuntu-2ubuntu122.04 (mixed mode, sharing)

LOCAL
- OS:macOS 13.1 (aarch64)
- Java runtime: Oracle OpenJDK Runtime Environment 17.0.2+8-86	
- Java VM:Oracle OpenJDK 64-Bit Server VM 17.0.2+8-86 (mixed mode, sharing)

**Comment by GianfrancoMS** [comment]
Applying normalization with these values worked:

```
normalization {
    runtimeClasspath {
        ignore '**/jdk/internal/jimage/**'
        ignore '**/jdk/internal/jrtfs/**'
        ignore '**/module-descriptor.txt'
        ignore '**/java/lang/invoke/**'
    }
}
```

**Comment by cdsap** [comment]
@GianfrancoMS, we have one interesting finding:
the cache miss for Java 17 is only present in versions 17.0.2, if you bump your Java version to 17.0.(3,4,5...) you will have a cache hit without normalization.

**Comment by GianfrancoMS** [comment]
Fixed in https://github.com/gradle/android-cache-fix-gradle-plugin/pull/419
